Source/Widgets/Viewer/Viewer.js

```diff
@@ -53,13 +54,73 @@ function onTimelineScrubfunction(e) {
   clock.shouldAnimate = false;
 }
 
+function getCesium3DTileFeatureDescription(feature) {
+  var propertyNames = feature.getPropertyNames();
+
+  var html = "";
+  propertyNames.forEach(function (propertyName) {
+    var value = feature.getProperty(propertyName);
+    if (defined(value)) {
+      html += "<tr><th>" + propertyName + "</th><td>" + value + "</td></tr>";
+    }
+  });
+
+  if (html.length > 0) {
+    html =
+      '<table class="cesium-infoBox-defaultTable"><tbody>' +
+      html +
+      "</tbody></table>";
+  }
+
+  return html;
+}
+
+function getCesium3DTileFeatureName(feature) {
+  var i;
+  var possibleNames = [];
+  var propertyNames = feature.getPropertyNames();
+  for (i = 0; i < propertyNames.length; i++) {
+    var propertyName = propertyNames[i];
+    if (/^name$/i.test(propertyName)) {
+      possibleNames[0] = feature.getProperty(propertyName);
+    } else if (/name/i.test(propertyName)) {
+      possibleNames[1] = feature.getProperty(propertyName);
+    } else if (/^title$/i.test(propertyName)) {
+      possibleNames[2] = feature.getProperty(propertyName);
+    } else if (/^(id|identifier)$/i.test(propertyName)) {
+      possibleNames[3] = feature.getProperty(propertyName);
+    } else if (/element/i.test(propertyName)) {
+      possibleNames[4] = feature.getProperty(propertyName);
+    } else if (/(id|identifier)$/i.test(propertyName)) {
+      possibleNames[5] = feature.getProperty(propertyName);
```

Comment:
Would it be simpler to do `possibleNames.push` in each one here?

Reply:
I meant to add a code comment here, but the idea is that we need to iterate all property names to find potential candidates, but since we prefer some property names over others, we store them in an indexed array and then use the first defined element in the array as the preferred choice. i.e. if we find both `name` and `element`, prefer `name`.